ui/app/mainui/AppMain.qml

```diff
@@ -224,6 +239,84 @@ Item {
+            id: dlgConfirmExternalLink
+
+            required property string link
+            required property string domain
+
+            width: 521
+            destroyOnClose: true
+
+            header: StatusDialogHeader {
+                headline.title: qsTr("Before you go")
+                actions.closeButton.onClicked: dlgConfirmExternalLink.close()
+                leftComponent: StatusRoundIcon {
+                    asset.name: "browser"
+                    asset.isImage: true
+                }
+            }
+            contentItem: ColumnLayout {
+                spacing: 20
+                StatusBaseText {
+                    Layout.fillWidth: true
+                    wrapMode: Text.WrapAtWordBoundaryOrAnywhere
+                    text: qsTr("This link is taking you to the following site. Be careful to double check the URL before you go.")
+                }
+                Rectangle {
+                    Layout.fillWidth: true
+                    Layout.preferredHeight: 66
+                    radius: Style.current.halfPadding
+                    color: Theme.palette.baseColor4
+
+                    StatusBaseText {
+                        anchors.fill: parent
+                        anchors.leftMargin: Style.current.padding
+                        anchors.rightMargin: Style.current.padding
+                        anchors.topMargin: 11
+                        anchors.bottomMargin: Style.current.halfPadding
+                        text: dlgConfirmExternalLink.link
+                        wrapMode: Text.WrapAnywhere
+                        elide: Text.ElideRight
+                    }
+                }
+                StatusCheckBox {
+                    id: trustDomainCheckbox
+                    Layout.fillWidth: true
+                    text: qsTr("Trust <b>%1</b> links from now on").arg(dlgConfirmExternalLink.domain)
+                }
+            }
+            footer: StatusDialogFooter {
+                rightButtons: ObjectModel {
+                    StatusFlatButton {
+                        text: qsTr("Cancel")
+                        onClicked: dlgConfirmExternalLink.close()
+                    }
+                    StatusButton {
+                        text: qsTr("Visit site")
+                        onClicked: {
+                            console.warn("!!! WHITELIST:", JSON.stringify(localAccountSensitiveSettings.whitelistedUnfurlingSites))
+                            console.warn("!!! CURRENT DOMAIN WHITELISTED:", !!localAccountSensitiveSettings.whitelistedUnfurlingSites[dlgConfirmExternalLink.domain])
+
+                            // (optionally) save the domain to whitelist
+                            if (trustDomainCheckbox.checked) {
+                                console.warn("!!! SAVE DOMAIN TO WHITELIST:", dlgConfirmExternalLink.domain)
+                                RootStore.updateWhitelistedUnfurlingSites(dlgConfirmExternalLink.domain, true)
```

Comment:
I don't think it's a good idea to reuse the old local white-list
- it might contain some items already (though the meaning of this list is quite common, it was a bit different thing in the first place, so I'd say it would be good to start it from scratch)
- if we use it, then we will have to migrate it later (if we are good guys :D)

I discussed it with @jrainville before seeing your PR. We thought that we could just hide this trust checkbox for now, until https://github.com/status-im/status-go/issues/4132 is implemented.

wdyt?

Reply:
Well yeah... been thinking about this too but I have exactly the opposite idea: we should reuse the existing list just because it's a logical extension of what we have already: list of trusted sites that we also provide previews for. In the end, when https://github.com/status-im/status-go/issues/4132 is done, we could start syncing that local setting to other devices too. 

> We thought that we could just hide this trust checkbox for now, until https://github.com/status-im/status-go/issues/4132 is implemented.

I don't know... it will be super annoying to confirm every link click if it's hidden.

Reply:
> Well yeah... been thinking about this too but I have exactly the opposite idea: we should reuse the existing list just because it's a logical extension of what we have already: list of trusted sites that we also provide previews for.

If the migration is easy, then yea, it's a good idea to reuse it, since it helps users reduce the amount of work they have to do.
The reason why using a completely different list is possible is because the UI for it in the settings will be completely different in the end.

> I don't know... it will be super annoying to confirm every link click if it's hidden.

It would be temporary until it's implemented. I'd say for public release it should be done, but it might just not be in 0.15.

Also, this popup should only be when clicking on an unfurl. Clicking on the link directly doesn't need the popup.
The reason is that when you click on the unfurl, you don't see the URL, so you can't confirm what you are clicking on (someone could create an unfurl image that says it's Youtube, but in the end it's actually a phishing site).

Reply:
I see pros and cons in both solution and personally tend to not reuse the old one, but I'm totally fine with both ways 🙂 It's not really of big deal anyway.

> we could start syncing that local setting to other devices too.

Hmm, I don't think it's will be simple to migrate `QSettings` to our database table (to be able to sync them).

Reply:
Alright, so I'll change this PR to:
- use a separate whitelist from the existing one, therefore keeping the checkbox functionality (could even be a local QML `Settings` in the end, if it's only temporary and we don't need the NIM part here)
- use it only for the unfurled URLs, and not for all links

wdyt @jrainville @igor-sirotin ?